predict.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+import time
+import ml_metrics as metrics
+import argparse
+from sklearn import linear_model
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.learning_curve import learning_curve
+import matplotlib.pyplot as plt
+from sklearn import cross_validation
+import numpy as np
+from termcolor import colored
+from sklearn.cross_validation import ShuffleSplit
+import pickle
+
+def parse(line):
+    tokens = line.split(',')
+    features = [tokens[3], tokens[4], tokens[5], tokens[13], tokens[14], tokens[15], tokens[16], 
+        tokens[17], tokens[20], tokens[21], tokens[22]]
+    features = [float (f) for f in features]
+    label = int (tokens[23])
+    return (features, label)
+
+def mean_average_precision(predicted_probability_distribution, 
+                           list_of_classes, 
+                           test_y):
+    total_error = 0
+    k = -1
+    for row in predicted_probability_distribution:
+        k += 1
+        prob_of_target_class = 0
+        place = 1
+        assert len(list_of_classes) == len(row)
+        for i in xrange(0, len(row)):
+            if list_of_classes[i] == test_y[k]:
+                prob_of_target_class = row[i]
+                break
+        for i in xrange(0, len(row)):
+            if (row[i] > prob_of_target_class 
+                or (row[i] == prob_of_target_class and list_of_classes[i] > test_y[k])):
+                place += 1
+                if place >= 6: break
+        if place < 6:
+            total_error += 1.0 / place
+    return total_error / len(predicted_probability_distribution)
+
+def plot_learning_curve(estimator, title, X, y, ylim=None, cv=None,
```

Comment:
Please remove this learning curve code from here so this python script serve single purpose.